code/code/disc/disc_wrath.cc

```diff
@@ -15,103 +15,102 @@
 
 int plagueOfLocusts(TBeing *caster, TBeing *victim, int level, short bKnown)
 {
-  TMonster *locusts;
-  affectedData aff; 
-  int rc, swarm;
-  followData *k, *n;
-
   if (!caster->outside()) {
     act("Nothing seems to happen.", FALSE, caster, NULL, NULL, TO_ROOM);
     caster->sendTo("You need to be outside to summon locusts!\n\r");
     return SPELL_FAIL;
   }
+
   if (Weather::getWeather(*caster->roomp) == Weather::RAINY) {
     act("Nothing seems to happen.", FALSE, caster, NULL, NULL, TO_ROOM);
     caster->sendTo("How do you expect to summon locusts in the rain!\n\r");
     return SPELL_FAIL;
   }
+
+  if ((caster->followers) && (level < GOD_LEVEL1))  {
+    followData *currentFollower, *nextFollower;
+    for (currentFollower = caster->followers; currentFollower; currentFollower = nextFollower) {
+      nextFollower = currentFollower->next;
+      if (isname("locust", currentFollower->follower->name)) {
+        act("Nothing seems to happen.", FALSE, caster, NULL, NULL, TO_ROOM);
+        caster->sendTo("You would have to be a god to summon another locust swarm!\n\r");        
+        return SPELL_FAIL;
+      }
+    }
+  }
+
+  int vnumToLoad;
   if (level < 26) 
-    swarm = Mob::LOCUSTS25;
+    vnumToLoad = Mob::LOCUSTS25;
   else if (level < 31) 
-    swarm = Mob::LOCUSTS30;
+    vnumToLoad = Mob::LOCUSTS30;
   else if (level < 36) 
-    swarm = Mob::LOCUSTS35;
+    vnumToLoad = Mob::LOCUSTS35;
   else if (level < 41) 
-    swarm = Mob::LOCUSTS40;
+    vnumToLoad = Mob::LOCUSTS40;
   else 
-    swarm = Mob::LOCUSTS50;
+    vnumToLoad = Mob::LOCUSTS50;
 
-  if (!(locusts = read_mobile(swarm, VIRTUAL))) {
+  TMonster *locusts = read_mobile(vnumToLoad, VIRTUAL);
+
+  if (!locusts) {
     vlogf(LOG_BUG, "Spell PLAGUE_LOCUSTS unable to load mob...");
     caster->sendTo("Unable to create the locusts, please report this.\n\r");
     act("Nothing seems to happen.", FALSE, caster, NULL, NULL, TO_ROOM);
     return SPELL_FAIL;
-  }
-
-  if ((caster->followers) && (level < GOD_LEVEL1))  {
-    for (k = caster->followers; k; k = n) {
-      n = k->next;
-      if (isname("locust", k->follower->name)) {
-        act("Nothing seems to happen.", FALSE, caster, NULL, NULL, TO_ROOM);
-        caster->sendTo("You would have to be a god to summon another locust swarm!\n\r");
-        delete locusts;
-        locusts=NULL;
-        return SPELL_FAIL;
-      }
-    }
-  }
+  }  
 
-  level = min(level, 75);
+  float normalizedLevel = max(1.0, min(level, 75) / 2.0);
```

Comment:
`min(level, 75)` is always `level`, since our levels are <= 60.

Reply:
`level = caster->getSkillLevel(SPELL_PLAGUE_LOCUSTS);`

It's the learnedness or the level specified on the item if it's a wand or something.

Reply:
Ooh. The variable name could be more explicit, then.

Reply:
I believe this code gets called when a mob uses this prayer as well, but I'd have to confirm that. But if so, is it possible this code is there to limit the strength when used by very high level cleric mobs? 

I wasn't entirely sure why they coded it this way originally - my primary intention behind this specific change was just to avoid mutating the level parameter in that way, since the result was only used in a single place in the code afterwards.

Reply:
I forced a lvl 89 cleric mob to cast plague of locusts on another mob and confirmed in gdb that this code triggered from the mob prayer and forced its level down to 37.5. Still not really sure what the original intent was here.

@cizra How would you like me to proceed?

Reply:
At this point I'm just getting rid of this level adjustment. It doesn't seem to serve much purpose except to gimp a spell that's really not that powerful in the first place. If you guys object I'm happy to change it back, though.

Reply:
`min(level, 75)` When this is called as I posted it's the learnedness of the spell. Not the level of the caster. Looks like the intent is for level to be capped at 37.5 but even without the min() statement the maximum level that could be passed into the function is 100 which would be divided by 2.

If you want to rework it then you should probably rework what's passed in so we can have high level mobs summon terrible demon locusts that breath fire on puny mortals.

If you just leave level that's passed in as the level of the mobs we are going to have some bad assed clerics walking around.

Reply:
After further testing I'm realizing it's probably stupid to allow clerics to easily summon a mob the same level as themselves that also has fairly strong damage resistance. So I'm going to ask for some feedback in #balance on Discord.

Reply:
Addressed in https://github.com/sneezymud/sneezymud/pull/353/commits/bb6987a12f3deb50ecb985e9095d18fe80652d58